README.md

```diff
@@ -19,8 +35,148 @@ Install Docker and look at the Docker instructions.
+docker-compose run ci bundle exec rails c
+
+docker-compose exec ci sh     # attach to an existing container named ci
+                              # and run the sh into it
+
+docker-compose ps             # see listing of all the containers running
+                              # and there status
+
+docker-compose down           # Stop all container
+
+docker-compose build          # Trigger a build of all container
+                              # to be used each time you modify Dockerfile
+docker-compose up --build     # ditto but, also trigger a 'up' action
+
+# Deploying a new version
+
+## Deploy manually to kubernetes ( for reference only )
+
+At this point travis is configured for continuous deployment
+Each new build of branch 'master' triggers a deploy to google
+container engine ( aka GKE ).
+
+See .travis.xml and travis/continuous_deployment
+
+### Install gcloud SDK
+https://cloud.google.com/sdk/docs/quickstart-mac-os-x
+
+### Install kubectl
+gcloud components update kubectl
+
+### Point to GKE cluster
+gcloud container clusters get-credentials <GKE-cluster-name> \
+    --zone <google-cloud-zone> --project <google-project>
+
+You can visualize your kubectl config with 'kubectl config view'
+to see how kubectl config was updated to point to your cluster.
+Alternatively you could fetch the info manually from the cloud.google.com
+and use 'kubectl set' to create that config.
+
+### Build the container and assign it a tag containing a unique build-number
+docker build -t gcr.io/<google-project>/ci:<build-number> -f Dockerfile.upgrade_analyzer .
+
+### Push your local container to GCE so it can be consumed by Kubernetes
+gcloud docker push gcr.io/<google-project>/ci:<build-number>
+
+### Create and store the secret on GKE -- initial setup only
+kubectl create secret generic upgrade-analyzer-secrets --from-literal=github-token=YOUR_GITHUB_TOKEN
+
+### Deploy declaratively -- initial setup only
+kubectl create -f kubectl_config/upgrade-analyzer-deployment.yml
+kubectl edit deployment/upgrade-analyzer-deployment
+# ^^ replace __REPO_NAME__ with with the repo you want to monitor
+#    replace __GOOGLE_PROJECT__ with the name of your GCE project name
+#    replace __BUILD_NUMBER__ with the build-number tagged container you
+                              just pushed
+
+### Confirm pods is healthy
+kubectl get pods
+# ^^ make sure pod is 'running'
+kubectl edit deployment/upgrade-analyzer-deployment
+# ^^ make sure were running intended version
+
+### Deploy declaratively -- way to deploy after initial deploy
+kubectl set image deployment/upgrade-analyzer-deployment \
+  upgrade-analyzer=gcr.io/<google-project>/ci:<build-number>
+# ^^ deploy will not occur if you don't set a new image build-number
+#    if you want to force a pods recreation with same ci:<buildnumber>
+#    simply delete the running pods with 'kubectl delete pods POD_NAME'
+#    the replication controller will automatically recreate a new one
```

Comment:
The formatting of this file is still not what  is needed to be a properly formatted Markdown document.  These level-one headings look like they were probably meant to be code comments. This is our README and is front and center for anyone that stumbles across this project, so we should clean this up.